front_end/panels/timeline/TimelinePanel.ts

```diff
@@ -420,6 +460,55 @@ export class TimelinePanel extends UI.Panel.Panel implements Client, TimelineMod
         Extensions.ExtensionServer.Events.TraceProviderAdded, this._appendExtensionsToToolbar, this);
     SDK.TargetManager.TargetManager.instance().addEventListener(
         SDK.TargetManager.Events.SuspendStateChanged, this._onSuspendStateChanged, this);
+
+    // COHERENT BEGIN
+
+    this._traceLevelsDefinitions = [];
+    this._traceSystemsDefinitions = [];
+    const mainTarget = (SDK.TargetManager.TargetManager.instance().mainTarget() as SDK.Target.Target);
+    if (UIDevtoolsUtils.isUiDevTools()) {
+      this._controller = new UIDevtoolsController(mainTarget, this);
+    } else {
+      this._controller = new TimelineController(mainTarget, this);
+    }
+
+    this._controller?._tracingManager?.fetchTraceSystemsAndLevels().then((res) => {
+      this._traceSystemsDefinitions = res.systems;
+      this._traceLevelsDefinitions = res.levels;
+
+      this._controller?.dispose();
+      this._controller = null;
```

Comment:
This should break the world when you receive the response while tracing - e.g. when you refresh and trace a page (not sure if it occurs, but seems possible.

What if instead we get the tracing agent directly, as we seem to construct the controller for it.
Maybe through those calls https://github.com/CoherentLabs/devtools-frontend-fork/blob/13706efede840af5ccdc8ed8d62b38d4d310ab63/front_end/core/protocol_client/InspectorBackend.ts#L733

The following seems okay at a first glance, do you find issues with it?
```JS
fetchTraceSystemsAndLevels(agent) {
    const response = await agent.invoke_getTraceSystemsAndLevels();
    return response;
}

fetchTraceSystemsAndLevels(tracingAgent()).then(res => {
  // setup stuff
});
```

Reply:
sure, I reworked the tracing agent acquisition logic based on the suggestion and it seems fine. I've tested the new logic and everything works correctly (refreshing during tracing leads to no crashes) .